.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,41 @@
+name: "Code scanning - action"
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 14 * * 5'
+
+jobs:
+  CodeQL-Build:
+
+    runs-on: windows-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        # We must fetch at least the immediate parents so that if this is
+        # a pull request then we can checkout the head.
+        fetch-depth: 2
+
+    # If this run was triggered by a pull request event, then checkout
+    # the head of the pull request instead of the merge commit.
+    - run: git checkout HEAD^2
+      if: ${{ github.event_name == 'pull_request' }}
+      
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+
+    - name: Setup .NET Core
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 3.1.102
+    
+    - name: Build with dotnet
+      run: | 
+        pwsh -command ./build.ps1
```

Comment:
We need to do some bootstrapping (platyPS in addition to the .Net Core SDK) before we can build, which would possible also make the `setup-dotnet@v1` redundant. `pwsh -command` doesn't support the usage of semicolons unfortuately
```pwsh
Import-Module .\tools\appveyor.psm1
Invoke-AppveyorInstall -SkipPesterInstallation
./build.ps1 -All
```

Reply:
I made a couple of attempts to fix this feedback, and I could not get it to play nice with codeql. The build seems to be passing and uploading analysis without these steps, though.